zulipterminal/model.py

```diff
@@ -73,7 +73,7 @@ def __init__(self, controller: Any) -> None:
         self.msg_view = None  # type: Any
         self.msg_list = None  # type: Any
         self.narrow = []  # type: List[Any]
-        self.update = False
+        self.update_allowed = False
```

Comment:
Reading the code, this variable is only used inside `Model`, and only between startup and the first `get_messages`?
Therefore, if we just renamed this, I would suggest something more along the lines of `__have_fetched_initial_messages`?

However, if my logic is correct: in core.py, `Model` must finish initializing (including `_update_initial_data(...)`, whose threads, including the initial call to `get_messages`, must join and finish), before the `Model.poll_for_events()` starts.

if that is correct, we could set a flag at the end of `Model.__init__` and assert on it at the start of `poll_for_events()` to ensure this doesn't change, or simply remove it entirely?

Perhaps @amanagr can shed more light on this; was this intended for another purpose?

Reply:
Also, while searching for the usages of `model.update` I found `core.py` has is setting `self.update` on the `Controller`, and this seems like a vestige of some old code, unless I'm missing something.

Reply:
@punchagan Agreed about that; I think I saw that a while back, but was focused on other refactoring at the time, and wanted to understand the data flow better before looking into it.

Is this flag perhaps intended as a way of ensuring the index isn't updated at the same time by `get_messages` and `append_message`, throughout the time ZT is running? (just checking the flag doesn't resolve this situation, of course, it just skips the append, which just "drops" the message, and it isn't added to the UI?)

Reply:
The purpose of `model.update` is to ensure that we only append message once all the messages have been fetched for a narrow and we used to reset update to false every time the narrow changed in controller. This seems to have been refactored to something else entirely. In `core.py` it should be `model.update` instead. @sumanthvrao can you fix this which you are at it?